File: Lab4/SalesEmployee.py
from Employee import Employee

class SalesEmployee(Employee):
    def __init__(self, id, firstName ,lastName ,email, contactNo , location ,birthdate , incentive , target , achievedTarget, basicSalary, hra):

        super().__init__(id, firstName, lastName, email,contactNo, location , birthdate ) 

        self.incentive=incentive
        self.target=target
        self.achievedtarget=achievedTarget
        self.basicSalary=basicSalary
        self.hra=hra

    def doWork(self):
        print(f"{self.firstName } is selling products and meeting clients.") 

    def computePay(self):
        totalSalary =self.basicSalary + self.hra
        if self.achievedtarget >= self.target:
            totalSalary += self.incentive
        return totalSalary           

File: HRApp/HR/SalesEmployee.js
//child class
const Employee = require("./Employee");
const IAppraisable = require("./Interfaces/IAppraisable");

class SalesEmployee extends Employee {
    constructor(
        id, firstName, lastName, email, contactNumber, location, birthDate, incentive, target, achievedTarget,
        basicSalary ,hra
    )  {

        //super means call employee constructor first 
        super(id, firstName, lastName, email, contactNumber, location, birthDate);
        this.incentive = incentive;
        this.target = target;
        this.achievedTarget = achievedTarget;
        this.basicSalary = basicSalary;
        this.hra = hra;
    }

   

    doWork() {
        console.log(`${this.firstName} is selling products and meeting clients.`);
    }

    computepay() {
         let totalSalary = this.basicSalary + this.hra;
            if (this.achievedTarget >= this.target) {   
                totalSalary += this.incentive;
            }       
        return totalSalary;
    }

    }
module.exports = SalesEmployee;
